src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -1,32 +1,124 @@
-import { EVENT_JS_READY } from '../common/logger/constants';
-import { mark } from '../../utils/performance';
-import messages from '../common/messages';
-
+import {
+    FIELD_IS_EXTERNALLY_OWNED,
+    FIELD_PERMISSIONS,
+    FIELD_PERMISSIONS_CAN_UPLOAD,
+    IS_ERROR_DISPLAYED,
+    ORIGIN_METADATA_SIDEBAR_REDESIGN,
+} from '../../constants';
 import './MetadataSidebarRedesign.scss';
 
+import { normalizeTemplates } from '../../features/metadata-instance-editor/metadataUtil';
+import { isUserCorrectableError } from '../../utils/error';
+
+import type { ElementsXhrError } from '../../common/types/api';
+import type { MetadataEditor, MetadataTemplate } from '../../common/types/metadata';
+import type { BoxItem } from '../../common/types/core';
+
 const MARK_NAME_JS_READY = `${ORIGIN_METADATA_SIDEBAR_REDESIGN}_${EVENT_JS_READY}`;
 
 mark(MARK_NAME_JS_READY);
 
-function MetadataSidebarRedesign() {
+function MetadataSidebarRedesign({ api, fileId, isFeatureEnabled, onError, selectedTemplateKey, templateFilters }) {
+    const [editors, setEditors] = useState<Array<MetadataEditor>>([]);
+    const [error, setError] = useState<MessageDescriptor>(undefined);
+    const [file, setFile] = useState<BoxItem>(undefined);
+    const [templates, setTemplates] = useState<Array<MetadataTemplate>>([]);
+
+    const onApiError = (err: ElementsXhrError, code: string) => {
+        const { status } = err;
+        const isValidError = isUserCorrectableError(status);
+        setError(messages.sidebarMetadataEditingErrorContent);
+        onError(error, code, { error, [IS_ERROR_DISPLAYED]: isValidError });
+    };
+
+    const fetchMetadataErrorCallback = (e: ElementsXhrError, code: string) => {
+        onApiError(e, code);
+
+        setEditors(undefined);
+        setError(messages.sidebarMetadataFetchingErrorContent);
+        setTemplates(undefined);
+
+        // onApiError(e, code);
+        // MetadataSidebar.js
+        // onApiError(e, code, {
+        //     editors: undefined,
+        //     error: messages.sidebarMetadataFetchingErrorContent,
+        //     templates: undefined,
+        // });
+    };
+
+    const fetchMetadataSuccessCallback = ({ fetchedEditors, fetchedTemplates }) => {
+        setEditors(fetchedEditors ?? []); // fetchedEditors and fetchedTemplates are undefined now
+        setError(undefined);
+        setTemplates(normalizeTemplates(fetchedTemplates ?? [], selectedTemplateKey, templateFilters));
+    };
+
+    const fetchMetadata = fetchedFile => {
+        if (!fetchedFile) return;
+
+        api.getMetadataAPI(false).getMetadata(
+            fetchedFile,
+            fetchMetadataSuccessCallback,
+            fetchMetadataErrorCallback,
+            isFeatureEnabled,
+            { refreshCache: true },
+        );
+    };
+
+    const fetchFileErrorCallback = (e: ElementsXhrError, code: string) => {
+        onApiError(e, code);
+        setError(messages.sidebarFileFetchingErrorContent);
```

Comment:
Should this be removed, as the same error is being set in `onApiError` anyway?

Reply:
I haven't noticed it's actually a different error message. In that case it will be overwriting the error that was set in `onApiError`.